File: Components/Strata.Hello.Service/main/typescript/RequestReply/IHelloService.ts
import {ICompletionStage} from "strata.foundation.core/Concurrent";
import {SayHelloRequest} from "./SayHelloRequest";
import {SayHelloReply} from "./SayHelloReply";

export interface IHelloService
{
    sayHelloAsync(request: SayHelloRequest): ICompletionStage<SayHelloReply>;
}

File: Components/Strata.Hello.Service/main/java/strata/hello/service/requestreply/IHelloService.java
//////////////////////////////////////////////////////////////////////////////
// IHelloService.java
//////////////////////////////////////////////////////////////////////////////

package strata.hello.service.requestreply;

import java.util.concurrent.CompletionStage;

public
interface IHelloService
{
    SayHelloReply
    sayHelloSync(SayHelloRequest request);

    CompletionStage<SayHelloReply>
    sayHelloAsync(SayHelloRequest request);
}

//////////////////////////////////////////////////////////////////////////////